source/extensions/filters/http/oauth2/filter.cc

```diff
@@ -93,6 +93,26 @@ authScopesList(const Protobuf::RepeatedPtrField<std::string>& auth_scopes_protos
   return scopes;
 }
 
+// Transforms the proto list into encoded resource params
+// Takes care of percentage encoding http and https is needed
+std::string encodeResourceList(const Protobuf::RepeatedPtrField<std::string>& resources_protos) {
+  std::string result = "";
+
+  if (!resources_protos.empty()) {
+    std::string h1 = "http://";
+    std::string h2 = "https://";
+
+    for (const auto& resource : resources_protos) {
+      if (resource.rfind(h1, 0) == 0 || resource.rfind(h2, 0) == 0) {
```

Comment:
Can we extract the URL detection logic into its own function?

Currently this allocates two strings every time we do this which seems unnecessary. How about we rewrite this as
`absl::StartsWith(resource, "http://") || absl::StartsWith(resource, "https://")`?